numba/tests/test_npdatetime.py

```diff
@@ -176,6 +176,39 @@ def test_combine_datetime_timedelta_units(self):
 DT = np.datetime64
 
 
+class TestMiscCompiling(TestCase):
+
+    def test_jit_explicit_signature(self):
+        def _check_explicit_signature(sig):
+            f = jit(sig, nopython=True)(add_usecase)
+            # Just a sanity check
+            args = DT(1, 'ms'), TD(2, 'us')
+            expected = add_usecase(*args)
+            self.assertPreciseEqual(f(*args), expected)
+
+        # Test passing the signature in object form
+        sig = types.NPDatetime('us')(types.NPDatetime('ms'), types.NPTimedelta('us'))
+        _check_explicit_signature(sig)
+        # Same with the signature in string form
+        sig = "NPDatetime('us')(NPDatetime('ms'), NPTimedelta('us'))"
+        _check_explicit_signature(sig)
+
+    def test_vectorize_explicit_signature(self):
+        def _check_explicit_signature(sig):
+            f = vectorize([sig], nopython=True)(mul_usecase)
+            # This isn't really right but we can't do better than this,
```

Comment:
When you say "this isn't really right" is it because the check should be checking against `TD(2, 's')` instead?